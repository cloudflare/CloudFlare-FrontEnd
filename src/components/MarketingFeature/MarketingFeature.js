import React, { Component, PropTypes } from 'react';

export default class MarketingFeature extends Component {
  render() {
    /*
         * These styles are stolen from the marketing site and aren't in our CSS
         */
    let divStyles = {
      padding: '30px 15px 30px 15px'
    };
    let iconStyles = {
      display: 'block',
      width: '40px',
      height: 'auto',
      maxWidth: '100px',
      margin: '0 auto'
    };
    let largeLinkStyles = {
      padding: '20px 0 0 0',
      textAlign: 'center',
      fontSize: '16px',
      color: '#2f7bbf',
      width: '100%',
      display: 'block'
    };
    let columnPStyles = {
      padding: '10px 0 0 0',
      fontSize: '12px',
      textAlign: 'center'
    };

    return (
      <div style={divStyles}>
        <img src={this.props.imgSrc} style={iconStyles} />
        <span style={largeLinkStyles} href="#">
          {this.props.title}
        </span>
        <p style={columnPStyles}>
          {this.props.description}
        </p>
      </div>
    );
  }
}

MarketingFeature.propTypes = {
  imgSrc: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
};
