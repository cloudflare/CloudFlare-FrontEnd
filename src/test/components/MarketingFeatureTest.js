import React from 'react';
import MarketingFeature
  from '../../components/MarketingFeature/MarketingFeature';
import renderer from 'react-test-renderer';

it('renders enabled correctly', () => {
  const component = renderer
    .create(
      <MarketingFeature
        imgSrc="imgSrc"
        title="title"
        description="description"
      />
    )
    .toJSON();
  expect(component).toMatchSnapshot();
});
