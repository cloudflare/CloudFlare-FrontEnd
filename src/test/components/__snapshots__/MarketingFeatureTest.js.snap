// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders enabled correctly 1`] = `
<div
  style={
    Object {
      "padding": "30px 15px 30px 15px",
    }
  }
>
  <img
    src="imgSrc"
    style={
      Object {
        "display": "block",
        "height": "auto",
        "margin": "0 auto",
        "maxWidth": "100px",
        "width": "40px",
      }
    }
  />
  <span
    href="#"
    style={
      Object {
        "color": "#2f7bbf",
        "display": "block",
        "fontSize": "16px",
        "padding": "20px 0 0 0",
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    title
  </span>
  <p
    style={
      Object {
        "fontSize": "12px",
        "padding": "10px 0 0 0",
        "textAlign": "center",
      }
    }
  >
    description
  </p>
</div>
`;
